### ISSUE TITLE
Deleting of a  netixlan objects should lead to a complete removal from the DB

### ISSUE THREAD

**Issue by arnoldnipper** [peeringdb/peeringdb#1017]
**Describe the bug**
Currently, deleting a `netixlan` only means that the object changes status to `deleted``. However, often this leads to severe issues as a new participant is not able to add an association to the IXP with the same IP address.

**To Reproduce**

The situation
```
{
  "id": 17181,
  "net_id": 4507,
  "ix_id": 53,
  "name": "LONAP: LON0",
  "ixlan_id": 53,
  "notes": "",
  "speed": 20000,
  "asn": 56478,
  "ipaddr4": "5.57.80.195",
  "ipaddr6": "2001:7f8:17::dc9e:2",
  "is_rs_peer": false,
  "operational": true,
  "created": "2014-03-03T00:00:00Z",
  "updated": "2018-05-11T00:00:21Z",
  "status": "deleted"
}

{
  "id": 13403,
  "net_id": 1034,
  "ix_id": 53,
  "name": "LONAP: LON0",
  "ixlan_id": 53,
  "notes": "Ip address None was claimed by other netixlan",
  "speed": 1000,
  "asn": 39757,
  "ipaddr4": "5.57.80.193",
  "ipaddr6": null,
  "is_rs_peer": false,
  "operational": true,
  "created": "2013-02-16T00:00:00Z",
  "updated": "2021-08-01T00:10:05Z",
  "status": "deleted"
}
```
LONAP issued 5.57.80.193 and 2001:7f8:17::dc9e:2 to ASN 56478. And as LONAP also is providing an IX-F JSON the system even suggested adding this entry. 

**Expected behavior**

A participant should be able to add a connection to an IXP regardless of which ASN had used the addresses before.

**Who is affected by the problem?**

users and AC

**What is the impact?**

Frustration on the user's side and additional work for AC

**Are there security concerns?**
no

**Are there privacy concerns?**
no

**What are the proposed actions?**
On deleting a netixlan, completely remove the object

**What is the proposed priority?**
normal
![image002](https://user-images.githubusercontent.com/16888757/128059441-03241e09-2115-4335-9ed5-4f234371bb41.png)

**Comment by arnoldnipper** [comment]
@peeringdb/ac and @peeringdb/pc, please discuss!

**Comment by leovegoda** [comment]
I don't know why the status is just changed to `deleted` instead of removing the object. This change sounds reasonable but would it break anything else?

**Comment by arnoldnipper** [comment]
> I don't know why the status is just changed to `deleted` instead of removing the object.

This is how PDB works. Objects are not really deleted, but the status only changes from `ok` to `deleted`. Only user records are deleted hard.

TBD whether also all other multi-id objects (`ixfac` and `netfac`) should be deleted hard. IMO it would make sense as those deleted objects also already caused confusion.

**Comment by peterhelmenstine** [comment]
Cascade delete +1

**Comment by Juimar** [comment]
+ 1 deleting a netixlan, completely remove the object

**Comment by Juimar** [comment]
On deleting a netixlan, completely remove the object
+1

**Comment by netravnen** [comment]
@arnoldnipper when #983 is finalized, would #1017 not become redundant?

**Comment by arnoldnipper** [comment]
> @arnoldnipper when #983 is finalized, would #1017 not become partially redundant?

no. IMO, the `netixlan` should be removed completely as it does not contain any useful information. Same holds for `ixfac` and `netfac`

**Comment by grizz** [comment]
I'm -1 on this until we can see exactly what's going on. Also keep in mind having this cascade delete could potentially turn into a huge database rewrite (I'm not sure without tracing through the code).

#983 should take care of this if the network has the "allow ix-f updates" box checked, if we changed policy to always allow the ix-f import to remove stale entries, this would be a quick easy fix.

Also mentioning that because that came up on the last @peeringdb/pc call as it would fix a few issues. In my opinion, fixing this would be better done that way, in both time spent to fix it and database robustness.

**Comment by arnoldnipper** [comment]
@grizz, while I see a benefit in keeping `ix`, `fac`, `net`, and `org` objects (and only modify their status), I don't see a benefit in keeping `ixfac`, `netfac`, and `netixlan` objects. If there are, please let me know.

**Comment by grizz** [comment]
It's all data, and relevant in some cases. The technical standpoint of not changing that is probably a bigger deal and something that's better left to a bigger rewrite if possible.

**Comment by arnoldnipper** [comment]
Of course, it's data.

But a deleted `netixlan` is of no use. But every now and then it causes big headache to users as well as to @peeringdb/ac. I'm happy to go into detail offline.

Same is with `netfac` where we still have the `local_asn` field which is a ugly remnant from v1.0. I.e. there are still deleted `netfac` objects where the value of `local_asn` does not match the `net_id`. And if AC tries to manipulate a `fac` object via the AC interface you get crazy.

**Comment by arnoldnipper** [comment]
Attached a list of broken `netfac` objects

[20230713-netfac-broken.csv](https://github.com/peeringdb/peeringdb/files/12034113/20230713-netfac-broken.csv)

**Comment by leovegoda** [comment]
Closing bug reports that are more than 1 year old because there has been significant code change

**Comment by leovegoda** [comment]
Sorry, @arnoldnipper. I misread your label from August 2021.

Perhaps we should schedule this for a discussion?

**Comment by arnoldnipper** [comment]
> Perhaps we should schedule this for a discussion?

Perhaps ... otoh, I understand @grizz argument not to hard delete anything from the DB. Instead of hard deleting, fixing those _broken_ records would also do. And iirc, we have a similiar issue when a network wants to re-add a connection to a facility.
